pkg/codegen/dotnet/gen.go

```diff
@@ -1979,6 +1979,13 @@ func genPackageMetadata(pkg *schema.Package, assemblyName string, packageReferen
 
 	files.add(assemblyName+".csproj", projectFile)
 	files.add("logo.png", logo)
+
+	// Generate version.txt
+	versionFile := "version.txt"
+	if _, exists := files[versionFile]; !exists && pkg.Version != nil {
+		files[versionFile] = []byte(pkg.Version.String())
```

Comment:
Is this to handle the case that the caller has already added a `version.txt`?

Reply:
That was my intention, specifically if someone passes a `version.txt` to `GeneratePackage` through `extraFiles`. 

BTW: this is unique for `.NET` because every other generator already has the associated file, and we are just slotting in the version.